src/traces/scatter/attributes.js

```diff
@@ -398,9 +398,67 @@ module.exports = {
         description: [
             'Sets the fill color.',
             'Defaults to a half-transparent variant of the line color,',
-            'marker color, or marker line color, whichever is available.'
+            'marker color, or marker line color, whichever is available.',
+            'If fillgradient is specified, fillcolor is ignored except for',
+            'setting the background color of the hover label, if any.'
         ].join(' ')
     },
+    fillgradient: extendFlat({
+        orientation: {
+            valType: 'enumerated',
+            values: ['radial', 'horizontal', 'vertical', 'none'],
+            dflt: 'none',
+            editType: 'style',
+            description: [
+                'Sets the orientation of the color gradient.',
+                'Defaults to *none*.'
+            ].join(' ')
+        },
+        start: {
+            valType: 'number',
+            editType: 'calc',
+            description: [
+                'Sets the gradient start value.',
+                'It is given as the absolute position on the axis determined by',
+                'the orientiation. E.g., if orientation is *horizontal*, the',
```

Comment:
I am not sure if we should tie these to trace axis orientation. What if we want to support these for `polar` and `smith` charts as well? Wondering using orientation displayed on the paper may be a better option?

Reply:
@alexcjohnson